core/src/main/java/org/polypheny/db/catalog/entity/logical/LogicalColumn.java

```diff
@@ -0,0 +1,231 @@
+
+    @Serialize
+    @SerializeNullable
+    public LogicalDefaultValue defaultValue;
+
+    public DataModel dataModel = DataModel.RELATIONAL;
+
+
+    public LogicalColumn(
+            @Deserialize("id") final long id,
+            @Deserialize("name") @NonNull final String name,
+            @Deserialize("tableId") final long tableId,
+            @Deserialize("namespaceId") final long namespaceId,
+            @Deserialize("position") final int position,
+            @Deserialize("type") @NonNull final PolyType type,
+            @Deserialize("collectionsType") final PolyType collectionsType,
+            @Deserialize("length") final Integer length,
+            @Deserialize("scale") final Integer scale,
+            @Deserialize("dimension") final Integer dimension,
+            @Deserialize("cardinality") final Integer cardinality,
+            @Deserialize("nullable") final boolean nullable,
+            @Deserialize("collation") final Collation collation,
+            @Deserialize("defaultValue") final LogicalDefaultValue defaultValue ) {
+        this.id = id;
+        this.name = name;
+        this.tableId = tableId;
+        this.namespaceId = namespaceId;
+        this.position = position;
+        this.type = type;
+        this.collectionsType = collectionsType;
+        this.length = length;
+        this.scale = scale;
+        this.dimension = dimension;
+        this.cardinality = cardinality;
+        this.nullable = nullable;
+        this.collation = collation;
+        this.defaultValue = defaultValue;
+    }
+
+
+    public AlgDataType getAlgDataType( final AlgDataTypeFactory typeFactory ) {
+        AlgDataType elementType;
+        if ( this.length != null && this.scale != null && this.type.allowsPrecScale( true, true ) ) {
+            elementType = typeFactory.createPolyType( this.type, this.length, this.scale );
+        } else if ( this.length != null && this.type.allowsPrecNoScale() ) {
+            elementType = typeFactory.createPolyType( this.type, this.length );
+        } else {
+            assert this.type.allowsNoPrecNoScale();
+            elementType = typeFactory.createPolyType( this.type );
+        }
+
+        if ( collectionsType == PolyType.ARRAY ) {
+            elementType = typeFactory.createArrayType( elementType, cardinality != null ? cardinality : -1, dimension != null ? dimension : -1 );
+        } else if ( collectionsType == PolyType.MAP ) {
+            elementType = typeFactory.createMapType( typeFactory.createPolyType( PolyType.ANY ), elementType );
+        }
+
+        return typeFactory.createTypeWithNullability( elementType, nullable );
+    }
+
+
+    public String getNamespaceName() {
+        return Catalog.snapshot().getNamespace( namespaceId ).orElseThrow().name;
+    }
+
+
+    public String getTableName() {
+        return Catalog.snapshot().rel().getTable( tableId ).orElseThrow().name;
+    }
+
+
+    @Override
+    public PolyValue[] getParameterArray() {
```

Comment:
ParameterArray is a terrible hack I have introduced to the stack years ago for the JDBC interface. I hope we can get rid of this soon...

Reply:
Agreed. 👍